cmd/controller/controller.go

```diff
@@ -0,0 +1,163 @@
+/*
+Copyright 2018 OpenEBS Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package controller
+
+import (
+	"errors"
+	"fmt"
+	"os"
+
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/rest"
+	"k8s.io/client-go/tools/clientcmd"
+
+	"github.com/golang/glog"
+	apis "github.com/openebs/node-disk-manager/pkg/apis/ndm/v1alpha1"
+	clientset "github.com/openebs/node-disk-manager/pkg/client/clientset/versioned"
+	"github.com/openebs/node-disk-manager/pkg/signals"
+	"github.com/openebs/node-disk-manager/pkg/util"
+)
+
+const (
+	NDMKind     = "Disk"                   // NDMKind is the CR kind.
+	NDMVersion  = "openebs.io/v1alpha1"    // NDMVersion is the CR version.
+	NDMHostKey  = "kubernetes.io/hostname" // NDMHostKey is host name label prefix.
+	NDMActive   = "Active"                 // NDMActive is constant for active resource status.
+	NDMInactive = "Inactive"               // NDMInactive is constant for inactive resource status.
+)
+
+// ProbeChannel is used to send a copy of controller object to each probe.
+// Each probe can get the copy of controller struct any time they
+// need to read the channel.
+var ProbeChannel = make(chan *Controller)
```

Comment:
Should we rename this to "ControllerBroadcastChannel" -- since this is mainly used for advertising the controller object to any of the listeners/receivers. It could be "probes" today and could be others tomorrow..

Reply:
Renamed the channel as `ControllerBroadcastChannel`